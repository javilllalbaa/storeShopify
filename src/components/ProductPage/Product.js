import "./styles.scss";
import img from "../../assets/images/sin-imagen.jpeg";

const Product = ({ product }) => {

  return (
    <div className="product-container">

      {
        product && product.map((att, index) => (
          <div className="product-content" key={index}>
            <div className="container-img">
              <img
                className="product-image product-responsive"
                src={att.image == null ? img : att.image.src}
                alt={att.handle}
              />
            </div>
            <div className="product-section product-responsive">
              <div className="product-id">
                Código de Catálogo: {att.id}
              </div>
              <h1 className="product-title">{att.title}</h1>
              <ul className="product-attributes">
                <li>Precio: {att.variants[0].price}</li>
                <li>Vendedor: {att.vendor}</li>
              </ul>
            </div>
          </div>
        ))
      }

    </div>
  );

};

export default Product