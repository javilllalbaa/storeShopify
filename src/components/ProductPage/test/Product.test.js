import React from "react";
import { shallow } from "enzyme";

import Product from "../Product";

describe("ProductPage component", () => {
  const testProps = {
    product: [
      {
          "id": 8005666078999,
          "title": "dawn snow",
          "body_html": null,
          "vendor": "test-options-javi",
          "product_type": "",
          "variants": [
              {
                  "id": 44054249144599,
                  "product_id": 8005666078999,
                  "title": "Default Title",
                  "price": "9.41"
              }
          ],
          "image": {
              "id": 39860528251159,
              "product_id": 8005666078999,
              "src": "https://cdn.shopify.com/s/files/1/0678/1684/9687/products/helados.jpg?v=1668597326"
          }
      }
  ]
  };
  describe("WHEN the Product component is render", () => {
    const wrapper = shallow(<Product {...testProps} />);

    it("THEN should display Product component", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
