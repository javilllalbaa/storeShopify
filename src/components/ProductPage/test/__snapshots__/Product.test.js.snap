// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPage component WHEN the Product component is render THEN should display Product component 1`] = `ShallowWrapper {}`;
